### PR TITLE
[FLOC-4327] Stop logging the response body of REST API requests

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4327

Copied from #2720 with some test alterations.

### CODE REVIEW

File: flocker/restapi/_logging.py [hunk 1]
```diff
@@ -36,5 +36,5 @@
 JSON_REQUEST = ActionType(
     LOG_SYSTEM + u":json_request",
     [JSON],
-    [RESPONSE_CODE, JSON],
+    [RESPONSE_CODE],
     u"A request containing JSON request and response bodies.")
```
Comment (zubron) [inline flocker/restapi/_logging.py]: ❓ Is this description still correct, should we be referring to the response body here?

**Comment by zubron** [comment]
Looks good, thanks @wallrj! Just one point about maybe updating an eliot action description. Please address if you think you need to and then merge.

**Comment by wallrj** [comment]
Thanks @zubron I fixed the description and added an explanatory comment. I'll merge when the builds pass.

**Comment by wallrj** [comment]
I'm ignoring the two cinder failures above. We're seeing them on master and in other branches. Probably due to Rackspace problems.

**Pull request merged**